WordPress/Classes/Services/ReaderTopicService+FollowedInterests.swift

```diff
@@ -0,0 +1,56 @@
+import Foundation
+
+// MARK: - ReaderFollowedInterestsService
+
+/// Protocol representing a service that retrieves the users followed interests/tags
+protocol ReaderFollowedInterestsService: AnyObject {
+    /// Fetches the users locally followed interests
+    /// - Parameter completion: Called after a fetch, will return nil if the user has no interests or an error occurred
+    func fetchFollowedInterestsLocally(completion: @escaping ([ReaderTagTopic]?) -> Void)
+
+    /// Fetches the users followed interests from the network, then returns the sync'd interests
+    /// - Parameter completion: Called after a fetch, will return nil if the user has no interests or an error occurred
+    func fetchFollowedInterestsRemotely(completion: @escaping ([ReaderTagTopic]?) -> Void)
+}
+
+// MARK: - CoreData Fetching
+extension ReaderTopicService: ReaderFollowedInterestsService {
+    public func fetchFollowedInterestsLocally(completion: @escaping ([ReaderTagTopic]?) -> Void) {
+        completion(followedInterests())
+    }
+
+    public func fetchFollowedInterestsRemotely(completion: @escaping ([ReaderTagTopic]?) -> Void) {
+        fetchReaderMenu(success: { [weak self] in
+            guard let `self` = self else {
```

Comment:
Whats the purpose of this guard? Can we just do: `self?.fetchFollowedInterestsLocally(completion: completion)` ?

Reply:
I originally had it like that, but if self fails nothing happens. This just ensures the callback is called no matter what.

Reply:
In what cases was self nil? Usually that would point to a sever issue

Reply:
If we're pretty sure that `self` will exist we can use `[unowned self]`.

Reply:
Changed here: 875e1a8567a3c87e244590a83e64327d3fcd601c